interview-questions.md

```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a function? Why would you use one?
 
   Your answer:
-
+    It is a block of code that performs a specific task
   Researched answer:
-
+    The function is invoked only when you call the function.
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
+    .map() is a higher order function that returns a new array where you have applied a function on the element and changes the original element.  And .filter() also returns a new array of the elements but does not change the original values.
```

Comment:
Kind of- Both do not alter the original array. When you say 'apply a function on an element' what do you mean by function?